src/main/kotlin/org/rust/ide/presentation/Utils.kt

```diff
@@ -61,19 +61,39 @@ fun getPresentationForStructure(psi: RsElement): ItemPresentation {
 }
 
 private fun buildTemplateImplTypeName(psi: RsImplItem): String? {
-    val paramsList = mutableListOf<String>()
-    val baseName = psi.typeReference?.baseType?.name ?: return null
-    for (typeParameter in psi.typeParameters) {
-        val name = typeParameter.name ?: continue
-        val typeBounds = typeParameter.typeParamBounds?.childrenOfType<RsPolybound>()
+
+    val populatePolyboundsMap = fun(type: String, typeBounds: List<RsPolybound>?, map: LinkedHashMap<String, MutableList<String>>): Unit {
```

Comment:
I wonder why `populatePolyboundsMap` needs to be a val of a fun instead of a normal fun?
Also `RsTypeParameter.bounds` seems to already implement all logic to extract bounds from both places. Can we reuse that instead of recoding it or is there a reason why it can't be used?

This would allow to simplify the code to something like:
```rust
    val boundsMap = LinkedHashMap<String, MutableList<String>>()
    val baseName = psi.typeReference?.baseType?.name ?: return null
    for (typeParameter in psi.typeParameters) {
        val name = typeParameter.name ?: continue

        val boundsList = mutableListOf<String>()
        typeParameter.bounds.mapNotNullTo(boundsList) { it.bound.traitRef?.path?.referenceName }
        if (boundsMap.containsKey(name))
            boundsMap[name]!!.addAll(boundsList)
        else
            boundsMap.put(name, boundsList)
    }
```
And in my (very quick) tests it yielded the same results

Reply:
```populateBoundsMap``` is used only in ```buildTemplateImplTypeName``` function and nowhere else, so I don't see any point in introducing it in higher level scope.

Thanks for pointing to ```RsTypeParameter.bounds``` - I will simplify code now. Also, ```populateBoundsMap``` becomes obsolete and will be removed.